src/nodes.jl

```diff
@@ -95,11 +113,72 @@ This is the most common `DispatchNode`.
 @auto_hash_equals type Op <: DispatchNode
     result::DeferredFuture
     func::Function
+    label::String
     args
     kwargs
 end
 
-Op(func::Function, args...; kwargs...) = Op(DeferredFuture(), func, args, kwargs)
+function Op(func::Function, args...; kwargs...)
+    Op(
+        DeferredFuture(),
+        func,
+        string(Symbol(func)),
+        args,
+        kwargs
+    )
+end
+
+"""
+    summary(op::Op)
+
+Returns a string representation of the `Op`
+with its label and the args/kwargs types.
+
+NOTE: if an arg/kwarg is a `DispatchNode` with a label
+it will be printed with that arg.
+"""
+function Base.summary(op::Op)
+    function val_summary(val)
+        result = split(string(typeof(val)), '.')[end]
```

Comment:
I see what you're trying to do here, but maybe we should just take off `Dispatcher.`? Otherwise we could clobber something useful.

I'm leaning towards not parsing off the `Dispatcher.` at all.